File: week-11/protoseo/신고 결과 받기.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int n = id_list.length;
        int[] answer = new int[n];
        int[] reportedCount = new int[n];
        Map<String, Integer> idByIndex = new HashMap<>();
        List<String>[] reportIdByIndex = new ArrayList[n];

        for (int i = 0; i < n; i++) {
            idByIndex.put(id_list[i], i);
            reportIdByIndex[i] = new ArrayList<>();
        }

        Set<String> reportSet = new HashSet<>(Arrays.asList(report));
        for (String r : reportSet) {
            String[] split = r.split(" ");
            reportIdByIndex[idByIndex.get(split[0])].add(split[1]);
            reportedCount[idByIndex.get(split[1])]++;
        }

        for (int i = 0; i < n; i++) {
            int count = 0;
            for (String id : reportIdByIndex[i]) {
                if (reportedCount[idByIndex.get(id)] >= k) {
                    count++;
                }
            }
            answer[i] = count;
        }
        return answer;
    }
}

File: week-11/dahyen0o/신고-결과-받기.py
from collections import defaultdict

def solution(id_list, report, k):
    reported = defaultdict(set)
    
    for r in report:
        fr, to = r.split()
        reported[to].add(fr)
    
    answer = defaultdict(int)
    
    for to, frs in reported.items():
        if len(frs) >= k:
            for fr in frs:
                answer[fr] += 1
        
    return [answer[id] for id in id_list]
